src/reference/mod.rs

```diff
@@ -357,6 +357,8 @@
 //! earliest matching regular expression in each iteration, until we have exhausted the entire
 //! string, or none of the regular expressions match.
 //!
+//! Note that a regular expression that matches an empty string results in an error.
```

Comment:
Oh right, we need this constraint for a termination guarantee, don't we?

Reply:
Indeed! Found this out the hard way...